Source/missiles.cpp

```diff
@@ -3680,7 +3680,7 @@ void ProcessRhino(Missile &missile)
 	}
 	UpdateMissilePos(missile);
 	Point newPos = missile.position.tile;
-	if (!IsTileAvailable(monster, newPos) || (monster.ai == MonsterAIID::Snake && !IsTileAvailable(monster, newPosSnake))) {
+	if (!IsTileAvailable(monster, newPos) || (monster.ai == MonsterAIID::Snake && (!IsTileAvailable(monster, newPosSnake) || missile._miAnimFrame == missile._miAnimLen))) {

```

Comment:
```suggestion
	if (!IsTileAvailable(monster, newPos) || (monster.ai == MonsterAIID::Snake && (!IsTileAvailable(monster, newPosSnake) || missile._miAnimFrame >= missile._miAnimLen))) {
```

This would better match similar logic in `player.cpp`, such as...

https://github.com/diasurgical/DevilutionX/blob/2d27363228c4701f12bb3a45da06f47c13dc3c7b/Source/player.cpp#L1322